syncd/SaiDiscovery.cpp

```diff
@@ -157,7 +159,10 @@ void SaiDiscovery::discover(
                  * create, we don't need to query this attribute.
                  */
 
-                //continue;
+                if (m_flags & SaiDiscoveryFlags::SkipDefaultEmptyAttributes)
+                {
+                    continue;
+                }
```

Comment:
this seems to affect all attributes on all objects, please limit this scope to only Port

Reply:
Not related to objec type. e.g. discovery after warm boot should be without this flag - attributes could be populated with objects before in old life.
If object is just created/cold boot/fast boot - no need to query most default value attributes.
It depends on the context saidiscovery used for, so user should pass a flag.

Reply:
your scope is to port objects, please limit to port

Reply:
Added a check to limit usage to port